tools/manifest/utils.py

```diff
@@ -55,16 +55,16 @@ def to_os_path(path):
 
 
 def git(path):
-    # type: (bytes) -> Optional[Callable[..., Text]]
+    # type: (bytes) -> Optional[Callable[..., bytes]]
     def gitfunc(cmd, *args):
-        # type: (bytes, *bytes) -> Text
+        # type: (bytes, *bytes) -> bytes
         full_cmd = ["git", cmd] + list(args)
         try:
-            return subprocess.check_output(full_cmd, cwd=path, stderr=subprocess.STDOUT).decode('utf8')
```

Comment:
So I thought this change was correct (or at least aligns with our general strategy of assuming process output is UTF8); are you still planning to land it or is it in a different PR?

Reply:
I am not sure if we should do that for git commands output. There are actually a lot of ramifications as you have seen (like having binary hashes for example) of switching to a Text output in this case.

I agree that the general strategy is to assume UTF8 output for process calls but for this particular case I think we can make an exception. Do you prefer instead to keep the decode() and then encode() as needed?

Reply:
I see. I think if we are taking the position that command output is text then we should be consistent with that. If we want to then represent hashes as the actual byte value rather than text we should convert down. But thinking about it they end up in the manifest JSON, so we have to represent them as text. So I think my earlier comment was wrong, sorry.

Reply:
So... can we land it as is?

Reply:
No, I think we should reapply the changes that you reverted so that the git output is converted to a string, and the hash is a string.